resources/packs/core/dns.go

```diff
@@ -29,15 +29,17 @@ func (d *mqlDomainName) init(args *resources.Args) (*resources.Args, DomainName,
 		(*args)["fqdn"] = fqdn
 	}
 
-	dn, err := domain.Parse(fqdn.(string))
-	if err != nil {
-		return nil, nil, err
-	}
+	if fqdn != nil {
```

Comment:
I think initializing the `domainName` resource without `fqdn` is confusing to users. Maybe we just add a specific check like:

```go
fqdnArg, ok := (*args)["fqdn"]
if !ok {
  if transport, ok := d.MotorRuntime.Motor.Provider.(*network.Provider); ok {
    fqdnArg = transport.FQDN
  }

  (*args)["fqdn"] = fqdnArg
}

fqdn := fqdnArg.(string)
if fqdn == "" {
   return nil, nil, errors.New("domainName resource requires fqdn argument")
}

dn, err := domain.Parse(fqdn)
...
 
```

Reply:
Wouldn't this still panic on `fqdn := fqdnArg.(string)` ?

Reply:
But the error makes sense,

so how about:

```
	fqdn, ok := (*args)["fqdn"]
	if !ok {
		if transport, ok := d.MotorRuntime.Motor.Provider.(*network.Provider); ok {
			fqdn = transport.FQDN
		}

		(*args)["fqdn"] = fqdn
	}

	if fqdn == nil {
    return nil, nil, errors.New("domainName resource requires fqdn argument")
	}
  dn, err := domain.Parse(fqdn.(string))
```

Reply:
Yes. Great point. Your error handling is better 🥳

Reply:
Changed and tested 

```
Query encountered errors:
failed to create resource 'domainName': domainName resource requires fqdn argument
domainName: no data available
```